projects/plugins/jetpack/modules/sharedaddy/sharing-service.php

```diff
@@ -942,6 +942,11 @@ function sharing_display( $text = '', $echo = false ) {
 		return $text;
 	}
 
+	// Prevent from rendering sharing buttons in block which is fetched from REST endpoint by editor
+	if ( defined( 'REST_REQUEST' ) && REST_REQUEST ) {
```

Comment:
@coder-karen @jeherve Just to confirm, we'd like to _always_ remove the share buttons in REST API responses?

If we wanted to sometimes include share buttons in the REST API response, an alternative approach might be:
1. Only include share buttons in the REST API response when share buttons are enabled for the post type.
2. Don't ever include share buttons in the REST API response when it's a dynamic block request.

Reply:
Given that the injected sharing buttons require enqueued JS to function further down in the chain, I'm not sure why anyone would want the buttons to show up in REST API responses 🤔 

This tested well for me, and I think it'd be safe to ship and observe for any issue requests that pop up.

Reply:
When the editor requests this block, it appends the `context` GET param set to the `edit` value. I could include it in the condition to address @danielbachhuber concerns. What do you think?

Reply:
That could work, assuming the main intention is to hide the sharing buttons in the editor. That would tie in with the original linked issue.

Perhaps determining which REST API conditions to include sharing buttons in could be for a separate PR, with this one just removing them from the editor. That said I'll be releasing 11.2 in a few hours so I'm not sure if any of this should be a blocker. Given that the issue itself (share button output in the editor) has been around for quite some time it may be better that I don't try to rush this in to 11.2, then we can make sure all cases are covered.

Reply:
Or I can just merge as is, like @samiff suggested we can look out for any issues as is, though we could perhaps refine what we do and don't want to include in a further PR. I'll do this then unless anything comes up in the next few hours.

Reply:
@coder-karen I've just added that additional check and retested it - it still works as expected.

Reply:
Yes it does. I'll merge this as is, then I think we can follow up in another PR should we need to be stricter with the REST API conditions.